WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt

```diff
@@ -366,9 +360,61 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
 
     private fun initViewModel() {
         viewModel = ViewModelProviders.of(this, viewModelFactory).get(ReaderPostDetailViewModel::class.java)
+        viewModel.uiState.observe(
+                viewLifecycleOwner,
+                Observer<ReaderPostDetailsUiState> { state ->
+                    header_view.updatePost(state.headerUiState)
+                }
+        )
+
+        viewModel.snackbarEvents.observe(viewLifecycleOwner, Observer {
+                it?.applyIfNotHandled {
+                    showSnackbar()
+                }
+            }
+        )
+
+        viewModel.navigationEvents.observe(viewLifecycleOwner, Observer {
+            it.applyIfNotHandled {
+                when (this) {
+                    is ReaderNavigationEvents.ShowPostsByTag -> {
+                            ReaderActivityLauncher.showReaderTagPreview(context, this.tag)
+                    }
+                    is ShowBlogPreview -> ReaderActivityLauncher.showReaderBlogOrFeedPreview(
+                            context,
+                            this.siteId,
+                            this.feedId
+                    )
+                    is ReaderNavigationEvents.ShowReaderComments,
+                    is ReaderNavigationEvents.ShowNoSitesToReblog,
+                    is ReaderNavigationEvents.ShowSitePickerForResult,
+                    is ReaderNavigationEvents.OpenEditorForReblog,
+                    is ReaderNavigationEvents.ShowBookmarkedTab,
+                    is ReaderNavigationEvents.ShowBookmarkedSavedOnlyLocallyDialog,
+                    is ReaderNavigationEvents.SharePost,
+                    is ReaderNavigationEvents.OpenPost,
+                    is ReaderNavigationEvents.ShowReportPost -> { // TODO: Handle nav events
```

Comment:
Tip: I prefer to keep my name or project name next to the "TODO" so I can easily search for all the remaining todos before the project is released. It's quite easy to forget about a todo and these generic "TODO" items are easy to miss. What do you think about it?

Reply:
Sure, will add next time. Thank you 🙇‍♀️.